src/core/io/platform/f3xx/SPIf3xx.cpp

```diff
@@ -241,23 +241,18 @@ void SPIf3xx::configureSPI(uint32_t baudRate, SPIMode mode, bool firstBitMSB) {
     }
 
     // configure the clock prescaler to the closest baudrate to the requested
-    if (baudRate >= SPI_MAX_BAUD) {
-        halSPI.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_2;
-    } else if (baudRate >= SPI_MAX_BAUD / 2) {
-        halSPI.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_4;
-    } else if (baudRate >= SPI_MAX_BAUD / 4) {
-        halSPI.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_8;
-    } else if (baudRate >= SPI_MAX_BAUD / 8) {
-        halSPI.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_16;
-    } else if (baudRate >= SPI_MAX_BAUD / 16) {
-        halSPI.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_32;
-    } else if (baudRate >= SPI_MAX_BAUD / 32) {
-        halSPI.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_64;
-    } else if (baudRate >= SPI_MAX_BAUD / 64) {
-        halSPI.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_128;
-    } else {
-        halSPI.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_256;
+    uint32_t prescaler = (HAL_RCC_GetHCLKFreq() / baudRate);
+    // Convert the prescaler number to the bit value incrementally (log2)
+    halSPI.Init.BaudRatePrescaler = SPI_BAUDRATEPRESCALER_2;
+    while (prescaler >>= 1) {
+        halSPI.Init.BaudRatePrescaler++;
+    }
+    // Max value is a prescaler of 256 or 0x07+1.
```

Comment:
> 0x07+1
What?

Reply:
A prescaler of 256 is 0x7 but due to the loop the value for the calculated prescaler will be one more (+1) this is then subtracted before shifting in the next step to bits 5:3 for the BR field.

Reply:
Some explanation like that would do nicely in a comment around here